src/core_modules/capture-core-utils/rulesEngine/commonUtils/normalizeRuleVariable.js

```diff
@@ -3,30 +3,33 @@ import log from 'loglevel';
 import isString from 'd2-utilizr/lib/isString';
 import { typeKeys } from '../constants';
 
-// Turns the internal representation of a program rule variable into its "canonical" format
-// (e.g. numbers represented as strings get converted to numbers)
-export const normalizeRuleVariable = (data: any, valueType: string) => {
-    const convertNumber = (numberRepresentation) => {
-        if (isString(numberRepresentation)) {
-            if (isNaN(numberRepresentation)) {
-                log.warn(`rule execution service could not convert ${numberRepresentation} to number`);
-                return null;
-            }
-            return Number(numberRepresentation);
+const convertNumber = (numberRepresentation) => {
+    if (isString(numberRepresentation)) {
+        if (isNaN(numberRepresentation)) {
+            log.warn(`rule execution service could not convert ${numberRepresentation} to number`);
+            return null;
         }
-        return numberRepresentation;
-    };
+        return Number(numberRepresentation);
+    }
+    return numberRepresentation;
+};
 
-    const convertString = (stringRepresentation: number | string): string => stringRepresentation.toString();
+const convertBoolean = (value) => {
+    if (isString(value)) {
+        return value === 'true';
+    }
+    return value;
```

Comment:
Hey, 
Does it make sense to add to the `convertBoolean` function the use case for the `TRUE_ONLY` ruleEffectConverter which should [always return true](https://github.com/dhis2/capture-app/pull/3089/files#diff-831e3acf07b2db7e576e336385b36e9a35318c468d6824c6f680d14fccf01b76L29)?
Thanks!

Reply:
I was the one writing `[typeKeys.TRUE_ONLY]: () => true` once upon a time, but realised that this was utterly wrong while working on this ticket.